packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationUtilities.m

```diff
@@ -0,0 +1,168 @@
+  } else if (UIInterfaceOrientationMaskPortraitUpsideDown & orientationMask) {
+    return UIInterfaceOrientationPortraitUpsideDown;
+  }
+  return UIInterfaceOrientationUnknown;
+}
+
++ (NSMutableDictionary *)inverted:(NSDictionary *)dictionary
+{
+  NSMutableDictionary *invertedDictionary = [NSMutableDictionary new];
+  [dictionary enumerateKeysAndObjectsUsingBlock:^(id key, id value, BOOL *stop) {
+      invertedDictionary[value] = key;
+  }];
+  return invertedDictionary;
+}
+
+# pragma mark - import/export
+
++ (NSDictionary *)orientationLockMap
+{
+  static NSDictionary *orientationLockMap = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    orientationLockMap = @{
+      @0 : @(UIInterfaceOrientationMaskAllButUpsideDown),
+      @1 : @(UIInterfaceOrientationMaskAll),
+      @2 : @(UIInterfaceOrientationMaskPortrait | UIInterfaceOrientationMaskPortraitUpsideDown),
+      @3 : @(UIInterfaceOrientationMaskPortrait),
+      @4 : @(UIInterfaceOrientationMaskPortraitUpsideDown),
+      @5 : @(UIInterfaceOrientationMaskLandscape),
+      @6 : @(UIInterfaceOrientationMaskLandscapeLeft),
+      @7 : @(UIInterfaceOrientationMaskLandscapeRight),
+      @10: @(UIInterfaceOrientationMaskAllButUpsideDown)
+    };
+  });
+  
+  return orientationLockMap;
+}
+
++ (NSDictionary *)orientationMap
+{
+  static NSDictionary *orientationMap = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    orientationMap = @{
+      @(UIInterfaceOrientationPortrait)           : @1,
+      @(UIInterfaceOrientationPortraitUpsideDown) : @2,
+      @(UIInterfaceOrientationLandscapeLeft)      : @3,
+      @(UIInterfaceOrientationLandscapeRight)     : @4,
+    };
+  });
+  
+  return orientationMap;
+}
+
++ (UIInterfaceOrientationMask)importOrientationLock:(NSNumber *)orientationLock
+{
+  return [[self orientationLockMap][orientationLock] integerValue] ?: 0;
+}
+
++ (NSNumber *)exportOrientationLock:(UIInterfaceOrientationMask)orientationMask
+{
+  static NSMutableDictionary *exportOrientationLockMap = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    exportOrientationLockMap = [self inverted:[EXScreenOrientationUtilities orientationLockMap]];
+    exportOrientationLockMap[@(UIInterfaceOrientationMaskAllButUpsideDown)] = @0; // UIInterfaceOrientationMaskAllButUpsideDown is default value
+  });
+  
+  return exportOrientationLockMap[@(orientationMask)] ?: @(8);
+}
+
++ (NSNumber *)exportOrientation:(UIInterfaceOrientation)orientation
+{
+  return [self orientationMap][@(orientation)] ?: @(UIInterfaceOrientationUnknown);
+}
+
++ (UIInterfaceOrientation)importOrientation:(NSNumber *)orientation
+{
+  static NSDictionary *exportOrientationMap = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    exportOrientationMap = [self inverted:[self orientationMap]];
+  });
+  return [exportOrientationMap[orientation] intValue] ?: UIInterfaceOrientationUnknown;
+}
+
++ (BOOL)isIPad
```

Comment:
this visually hurts my eyes, but I don't know if anything else would be better 😃